scripts/pre_commit_linter.py

```diff
@@ -1158,6 +1158,76 @@ def _check_docstrings(all_files):
     return summary_messages
 
 
+def _check_args_order(all_files):
+    """This function ensures that class args(self or cls)
+    comes first in the function definition.
+    """
+    print 'Starting arguments order check'
+    print '----------------------------------------'
+    wrong_ordself_message = ('The first arg of the function should be self')
+    wrong_ordcls_message = ('The first arg of the function should be cls')
+    summary_messages = []
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.py')]
+    failed = False
+    for filename in files_to_check:
+        with open(filename, 'r') as f:
+            file_content = f.readlines()
```

Comment:
The `ast` library can make this file reading and function identification much simpler. Here's a code snippet you could consider using to iterate over all of the functions in a file and access their args:
```
for filename in files_to_check:
    with open(filename, 'r') as f:
        ast_file = ast.walk(ast.parse(f.read()))
        func_defs = [n for n in ast_file if isinstance(n, ast.FunctionDef)]
        for func in func_defs:
            print func.name
            print function_node.args.args
```